runtime/interp.c

```diff
@@ -244,15 +243,12 @@ value caml_interprete(code_t prog, asize_t prog_size)
   initial_sp_offset = (char *) caml_stack_high - (char *) caml_extern_sp;
   initial_external_raise = caml_external_raise;
   caml_callback_depth++;
-  saved_pc = NULL;
 
   if (sigsetjmp(raise_buf.buf, 0)) {
     caml_local_roots = initial_local_roots;
     sp = caml_extern_sp;
     accu = caml_exn_bucket;
-    pc = saved_pc; saved_pc = NULL;
-    if (pc != NULL) pc += 2;
-        /* +2 adjustment for the sole purpose of backtraces */
+    pc = NULL;
```

Comment:
Good riddance, `saved_pc`.  But, then, `caml_stash_backtrace` is always called with `pc == NULL`, right?  In which case you should just remove the `pc` argument to `caml_stash_backtrace`.  Then, this `pc = NULL` becomes useless.

Reply:
As far as I understand, `caml_stash_backtrace` is still given a non-`NULL` value of `pc` when the interpreter is explicitly executing a `RAISE` or `RERAISE` instruction.

Reply:
One thing I can do is pushing `pc-1` in the interpreter when raising the exception, and not passing it to `caml_stash_backtrace`.

Reply:
I have done that in the current proposal.